web/src/app/schedules/temp-sched/TempSchedAddNewShift.tsx

```diff
@@ -0,0 +1,272 @@
+                component={UserSelect}
+                label='Select a User'
+                name='userID'
+              />
+            </Grid>
+            <Grid item xs={6}>
+              <FormField
+                fullWidth
+                component={ISODateTimePicker}
+                label='Shift Start'
+                name='shift-start'
+                fieldName='start'
+                min={edit ? value.start : now}
+                mapOnChangeValue={(value: string, formValue: Value) => {
+                  if (!manualEntry) {
+                    const diff = DateTime.fromISO(value).diff(
+                      DateTime.fromISO(formValue.start),
+                    )
+                    formValue.end = DateTime.fromISO(formValue.end)
+                      .plus(diff)
+                      .toISO()
+                  }
+                  return value
+                }}
+                timeZone={zone}
+                disabled={q.loading}
+                hint={isLocalZone ? '' : fmtLocal(value?.start)}
+              />
+            </Grid>
+            <Grid item xs={6}>
+              {manualEntry ? (
+                <FormField
+                  fullWidth
+                  component={ISODateTimePicker}
+                  label='Shift End'
+                  name='shift-end'
+                  fieldName='end'
+                  min={edit ? value.start : now}
+                  hint={
+                    <React.Fragment>
+                      {!isLocalZone && fmtLocal(value?.end)}
+                      <div>
+                        <ClickableText
+                          data-cy='toggle-duration-on'
+                          onClick={() => setManualEntry(false)}
+                          endIcon={<ToggleIcon />}
+                        >
+                          Configure as duration
+                        </ClickableText>
+                      </div>
+                    </React.Fragment>
+                  }
+                  timeZone={zone}
+                  disabled={q.loading}
+                />
+              ) : (
+                <FormField
+                  fullWidth
+                  component={NumberField}
+                  label='Shift Duration (hours)'
+                  name='shift-end'
+                  fieldName='end'
+                  float
+                  // value held in form input
+                  mapValue={(nextVal: string, formValue: Value) => {
+                    const nextValDT = DateTime.fromISO(nextVal)
+                    if (!formValue || !nextValDT.isValid) return ''
+                    return nextValDT
+                      .diff(DateTime.fromISO(formValue.start), 'hours')
+                      .hours.toString()
+                  }}
+                  // value held in state
+                  mapOnChangeValue={(nextVal: string, formValue: Value) => {
+                    if (!nextVal) return ''
+                    return DateTime.fromISO(formValue.start)
```

Comment:
I'll do all of this here- but I haven't really _done_ any of this work. Things like this could probably be handled in a separate PR that fixes specific logic, since this body of work is intended mainly for the visual tweaks